src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -87,16 +90,8 @@ bool ClusterTaskManager::SchedulePendingTasks() {
         break;
       }
 
-      if (node_id_string == self_node_id_.Binary()) {
-        // Warning: WaitForTaskArgsRequests must execute (do not let it short
-        // circuit if did_schedule is true).
-        bool task_scheduled = WaitForTaskArgsRequests(work);
-        did_schedule = task_scheduled || did_schedule;
-      } else {
-        // Should spill over to a different node.
-        NodeID node_id = NodeID::FromBinary(node_id_string);
-        Spillback(node_id, work);
-      }
+      NodeID node_id = NodeID::FromBinary(node_id_string);
+      ScheduleOnNode(node_id, work);
```

Comment:
this is another major change: we hide the scheduling on local node/remote node into this `scheduleOnNode` function.